lib/auth/types/AuthType.js

```diff
@@ -341,6 +341,22 @@ export default class AuthType {
                 }
             }
 
+            // when exchange key is changed for SAML, the authorization header becomes invalid and hence the authenticate fails for logged in user
+	    // this will check if the authorization header is valid and if not return false as the token is invalid
+	    authToken = request.headers.authorization;
+            if(authToken) {
+                if (authToken.startsWith("bearer ")){
+                    try {
+                        await request.auth.securitySessionStorage.authenticate({
+                            authHeaderValue: request.headers.authorization
+                        });
+                    } catch (error) {
```

Comment:
can we catch based on error type

Reply:
I think when I tested it was a generic error, but cannot properly recall.
But yeah, will look into this